utcp.c

```diff
@@ -1594,6 +1594,29 @@ int utcp_shutdown(struct utcp_connection *c, int dir) {
 	return 0;
 }
 
+// Closes all the opened connections
+int utcp_abort_all_connections(struct utcp *utcp) {
+	if(!utcp) {
+		return -1;
+	}
+
+	int total_connections = utcp->nconnections;
+
+	for(int i = 0; i < total_connections; i++) {
+		struct utcp_connection *c = utcp->connections[i];
+
+		if(c->recv) {
+      errno = 0;
+			c->recv(c, NULL, 0);
+    }
+
+		if(utcp_abort(c) != -1) {
```

Comment:
Hm, the problem with `utcp_abort()` is that it also invalidates the `utcp_connection` handle. That means it's invalid for the application to call any of its channel functions. I would avoid that. Basically, it should do what `utcp_abort()` does but not set `c->reapable = true`.

Reply:
So, should I call `utcp_abort()` and then set `c->reapable = false` ?

Reply:
Not if `c->reapable` was `true` to start with. I think it's better to make a function `utcp_reset()` that resets a connection but doesn't set `c->reapable = false`, then use that in `utcp_abort_connection()`, and make `utcp_abort()` just call `utcp_reset()` and then set `reapable` to `true`.